packages/webpack/mixin.core.js

```diff
@@ -11,17 +11,28 @@ const {
 
 const { Mixin } = require('@untool/core');
 
+const { Resolvable } = require('./lib/utils/resolvable');
+
 class WebpackMixin extends Mixin {
-  createAssetsMiddleware() {
-    const assetsMiddleware = require('./lib/middleware/assets');
-    const { config, assets, assetsByChunkName } = this;
-    return assetsMiddleware(config, { assets, assetsByChunkName });
+  constructor(...args) {
+    super(...args);
+    this.assets = new Resolvable();
+  }
+  loadPrebuiltAssets() {
+    const { serverDir, assetFile } = this.config;
+    const path = join(serverDir, assetFile);
+    return exists(path) ? require(path) : {};
   }
   loadPrebuiltMiddleware() {
     const { serverDir, serverFile } = this.config;
     const path = join(serverDir, serverFile);
     return exists(path) ? require(path) : (req, res, next) => next();
   }
+  createAssetsMiddleware() {
+    const assetsMiddleware = require('./lib/middleware/assets');
+    const { assets } = this;
+    return assetsMiddleware(assets);
```

Comment:
As per my comment above in the assets middleware: Would it make sense to somehow indicate, that the assets middleware will use `this.assets` only to read instead of read and write to it?
Maybe something like this could already clear things up?
```javascript
return assetsMiddleware({ getAssets: this.assets.then.bind(this.assets) });
```
Or `resolvedAssets` instead of `getAssets` or something similar.

Reply:
Please see above: https://github.com/untool/untool/pull/115#discussion_r213105035